subsys/usb/device/class/netusb/function_rndis.c

```diff
@@ -207,8 +207,13 @@ static struct __rndis {
 static uint8_t manufacturer[] = CONFIG_USB_DEVICE_MANUFACTURER;
 static uint32_t drv_version = 1U;
 
-static uint8_t tx_buf[NET_ETH_MAX_FRAME_SIZE +
-				sizeof(struct rndis_payload_packet)];
+/**
+ * Assumes MaxPacketsPerTransfer of 1 and 802.2 (ethernet) medium.
+ */
+#define RNDIS_BUF_SIZE (NET_ETH_MAX_FRAME_SIZE + sizeof(struct rndis_payload_packet))
+
+static uint8_t tx_buf[RNDIS_BUF_SIZE];
+static uint8_t rx_buf[RNDIS_BUF_SIZE];
```

Comment:
It is actually a waste of RAM. I have to think about it, it could be fixed more efficiently.

Reply:
It is a crude fix.

If there is a better way then it might be applied to ECM as well:

https://github.com/zephyrproject-rtos/zephyr/blob/ab2b77219638c780294d6de622ae9784f7381180/subsys/usb/device/class/netusb/function_ecm.c#L31

Reply:
ECM is different, here the code below is designed to handle it transaction by transaction, on the most platforms only the maximum of endpoint MPS will be used in rx_buf.
```c
if (net_pkt_write(rndis.in_pkt,
                  rx_buf + hdr_offset, len - hdr_offset)) {
```

Reply:
Maybe add a REVISE note abor rx_buf definition that it could be optimized.